ent/encryption/encryption.cc

```diff
@@ -823,6 +823,21 @@ class encryption_file_io_extension : public sstables::file_io_extension {
         });
         co_return sink;
     }
+
+    future<data_source> wrap_source(const sstables::sstable& sst, sstables::component_type type, data_source source) override {
+        switch (type) {
+        case sstables::component_type::Scylla:
+        case sstables::component_type::TemporaryTOC:
+        case sstables::component_type::TOC:
+            co_return source;
+        default:
```

Comment:
Do not use `default`, instead list all fall-through components explicitely, so if we add a new one, the developer doing that will be forced to consider what to do here.

Reply:
I prefer to fix both places. You can fixing existing code in another PR, no need to expand the scope of this one.

Reply:
https://github.com/scylladb/scylladb/issues/23724